src/AppInstallerCommonCore/PortableARPEntry.cpp

```diff
@@ -0,0 +1,151 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#include "pch.h"
+#include "winget/PortableARPEntry.h"
+#include "winget/Manifest.h"
+
+using namespace AppInstaller::Utility;
+
+namespace AppInstaller::Registry::Portable
+{
+    constexpr std::wstring_view s_UninstallRegistryX64 = L"Software\\Microsoft\\Windows\\CurrentVersion\\Uninstall";
+    constexpr std::wstring_view s_UninstallRegistryX86 = L"Software\\Wow6432Node\\Microsoft\\Windows\\CurrentVersion\\Uninstall";
+    constexpr std::wstring_view s_DisplayName = L"DisplayName";
+    constexpr std::wstring_view s_DisplayVersion = L"DisplayVersion";
+    constexpr std::wstring_view s_Publisher = L"Publisher";
+    constexpr std::wstring_view s_InstallDate = L"InstallDate";
+    constexpr std::wstring_view s_URLInfoAbout = L"URLInfoAbout";
+    constexpr std::wstring_view s_HelpLink = L"HelpLink";
+    constexpr std::wstring_view s_UninstallString = L"UninstallString";
+    constexpr std::wstring_view s_WinGetInstallerType = L"WinGetInstallerType";
+    constexpr std::wstring_view s_InstallLocation = L"InstallLocation";
+    constexpr std::wstring_view s_PortableTargetFullPath = L"TargetFullPath";
+    constexpr std::wstring_view s_PortableSymlinkFullPath = L"SymlinkFullPath";
+    constexpr std::wstring_view s_SHA256 = L"SHA256";
+    constexpr std::wstring_view s_WinGetPackageIdentifier = L"WinGetPackageIdentifier";
+    constexpr std::wstring_view s_WinGetSourceIdentifier = L"WinGetSourceIdentifier";

```

Comment:
Couldn't you put these inside of a map and then use the map as a dictionary later instead of using all of the if..else if..else statements in the ToString method?

Reply:
Created a map for these strings as suggested

Reply:
Added a macro to stamp out cases and use a switch statement. Removed map